contracts/contracts/oracle/PriceFeedPair.sol

```diff
@@ -0,0 +1,133 @@
+ */
+contract PriceFeedPair is AggregatorV3Interface {
+    using SafeCast for uint256;
+    using SafeCast for int256;
+    using StableMath for uint256;
+
+    // Fields to make it compatible with `AggregatorV3Interface`
+    uint8 public constant override decimals = 18;
+    string public constant override description = "";
+    uint256 public constant override version = 1;
+    address public immutable addressFeed0;
+    address public immutable addressFeed1;
+    bool public immutable reverseFeed0;
+    bool public immutable reverseFeed1;
+    uint8 internal immutable decimalsFeed0;
+    uint8 internal immutable decimalsFeed1;
+
+    error PriceFeedAddressError(address _address);
+    error PriceFeedsMatchError();
+
+    constructor(
+        address _addressFeed0,
+        address _addressFeed1,
+        bool _reverseFeed0,
+        bool _reverseFeed1
+    ) {
+        if (_addressFeed0 == address(0)) {
+            revert PriceFeedAddressError(_addressFeed0);
+        }
+        if (_addressFeed1 == address(0)) {
+            revert PriceFeedAddressError(_addressFeed1);
+        }
+        if (_addressFeed0 == _addressFeed1) {
+            revert PriceFeedsMatchError();
+        }
+
+        decimalsFeed0 = AggregatorV3Interface(_addressFeed0).decimals();
+        decimalsFeed1 = AggregatorV3Interface(_addressFeed1).decimals();
+        addressFeed0 = _addressFeed0;
+        addressFeed1 = _addressFeed1;
+        reverseFeed0 = _reverseFeed0;
+        reverseFeed1 = _reverseFeed1;
+    }
+
+    function _calculatePrice(int256 priceFeed0, int256 priceFeed1)
+        internal
+        view
+        returns (int256)
+    {
+        uint256 price0 = priceFeed0.toUint256().scaleBy(18, decimalsFeed0);
+
+        if (reverseFeed0) {
+            price0 = uint256(1e18).divPrecisely(price0);
+        }
+
+        uint256 price1 = priceFeed1.toUint256().scaleBy(18, decimalsFeed1);
+
+        if (reverseFeed1) {
+            price1 = uint256(1e18).divPrecisely(price1);
+        }
+
+        return price0.mulTruncate(price1).toInt256();
+    }
+
+    /**
+     * @notice This function exists to make the contract compatible
+     * with AggregatorV3Interface (which OETHOracleRouter uses to
+     * get the price).
+     **/
+    function latestRoundData()
+        external
+        view
+        override
+        returns (
+            uint80,
+            int256 price,
+            uint256,
+            uint256 updatedAt,
+            uint80
+        )
+    {
+        // slither-disable-next-line unused-return
+        (, int256 _price0, , uint256 updatedAt0, ) = AggregatorV3Interface(
+            addressFeed0
+        ).latestRoundData();
+        // slither-disable-next-line unused-return
+        (, int256 _price1, , uint256 updatedAt1, ) = AggregatorV3Interface(
+            addressFeed1
+        ).latestRoundData();
+        updatedAt = Math.min(updatedAt0, updatedAt1);
+        price = _calculatePrice(_price0, _price1);
```

Comment:
@sparrowDom @naddison36 Thinking about this, one potential concern I have is that `latestRound` of one feed can be way off from the other feed. Would it make sense to compare the `updatedAt` values and make sure both the updates aren't too far off?

Like if FXS<>USD was last updated 24h ago and ETH<>USD 1h ago, I don't think it'd be very reliable to use this price. Especially if there was too much of mark movement in the past 24h. More so, since this is a generic contract

Reply:
Another way to do this for FXS<>ETH pair would be just use Uniswap V3's TWAP (by doing something similar to what we do on AuraPriceFeed): <https://blog.uniswap.org/uniswap-v3-oracles>

Reply:
We've discussed this on discord and figured out that the errors of multiple oracle price feeds stack. E.g. 2% error and 0.5% error price feeds result in 2.5% error price feed. We are OK keeping it like that for now, but still important to consider.